### PR TITLE
Add parameter doc to sunpy/sun/sun.py

### PR DESCRIPTION
**Pull request by dstansby**
There is a lot of repetition here because all the methods take the same argument - I wasn't sure how to avoid just copy and pasting the same bit of text, so that's what I've done. If anyone else knows a better way to do this please let me know!

### CODE REVIEW

**Comment by Cadair** [comment]
I haven't tested it but I think this would work:

```python

PARAMETER_DOCS = """
Parameters
---------------
...
""""

def add_parameter_docs(f):
    f.__doc__+= PARAMETER_DOCS
    return f

@add_parameter_docs
def sun_location(t='now'):
    """
    This does a thing.

    """
```

**Comment by nabobalis** [comment]
I just did that locally. 

It looks a bit off but I can't figure out why. 

screenshot from 2017-12-13 13-54-44
vs
screenshot from 2017-12-13 13-54-49

File: sunpy/sun/sun.py [hunk 1]
```diff
@@ -48,16 +48,18 @@
...
+    t :  [ int, float, time_string, datetime ], optional
+        Time, which can be either time_string, int, or datetime object. Parsed
+        using `~sunpy.time.parse_time`. Defaults to the current time.
 
 """
 
+
 def add_parameter_docs(f):
-    f.__doc__ += PARAMETER_DOCS
+    if isinstance(f.__doc__, str):
```
Comment (Cadair) [inline sunpy/sun/sun.py]: this should be `six.string_types`

Comment (nabobalis) [reply sunpy/sun/sun.py]: OH YEAH IN THIS CRAZY WORLD

**Comment by nabobalis** [comment]
We should move the decorator somewhere else just incase we need it for the future.

**Comment by Cadair** [comment]
Yeah, we should refactor it into utils and let it take arguments. Maybe a prepend and append argument to the decorator? 

We should merge this then open a separate issue for it.

**Comment by nabobalis** [comment]
Looks like I messed up. 
```
/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-3.6/sunpy/sun/sun.py:docstring of sunpy.sun.heliographic_solar_center:6: SEVERE: Unexpected section title.

Parameters
----------
/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-3.6/sunpy/sun/sun.py:docstring of sunpy.sun.solar_north:6: SEVERE: Unexpected section title.

Parameters
----------
/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-3.6/sunpy/sun/sun.py:docstring of sunpy.sun.sunearth_distance:6: ERROR: Unexpected indentation.
/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-3.6/sunpy/sun/sun.py:docstring of sunpy.sun.sunearth_distance:7: SEVERE: Unexpected section title.
```

I will fix these soon(TM)

**Comment by nabobalis** [comment]
Doc builds without problems now.

**Pull request merged**